src/NuGet.Clients/NuGet.SolutionRestoreManager.Interop/IVsProjectRestoreInfoSource.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.SolutionRestoreManager
+{
+    /// <summary>
+    /// Represents a package restore service API for integration with a project system.
+    /// Implemented by the project-system.
+    /// </summary>
+    [ComImport]
+    [Guid("35AD5FF2-6AB7-48E9-BCC0-189042410FA6")]
+    public interface IVsProjectRestoreInfoSource
+    {
+        /// <summary>
+        /// Project Unique Name.
+        /// Must be equivalent to the name provided in the <see cref="IVsSolutionRestoreService3.NominateProjectAsync(string, IVsProjectRestoreInfo2, CancellationToken)"/> or equivalent.
+        /// </summary>
+        /// <remarks>Never <see langword="null"/>.</remarks>
+        string Name { get; }
+
+        /// <summary>
+        /// Whether the source needs to do some work that could lead to a nomination. <br/>
+        /// This method may be called frequently, so it should be very efficient.
+        /// </summary> 
+        bool HasPendingNomination();
```

Comment:
Why not a property? Properties tend to imply fast completion, in line with the doc comment here.

Reply:
Great point,. I'll fix it quickly before the APIs get on your end.